crates/uv-interpreter/src/discovery.rs

```diff
@@ -0,0 +1,675 @@
+use thiserror::Error;
+use uv_cache::Cache;
+use which::which;
+
+use crate::implementation::ImplementationName;
+use crate::managed::toolchains_for_current_platform;
+use crate::py_launcher::py_list_paths;
+use crate::virtualenv::virtualenv_python_executable;
+use crate::virtualenv::{virtualenv_from_env, virtualenv_from_working_dir};
+use crate::{Interpreter, PythonVersion};
+use std::borrow::Cow;
+
+use std::collections::HashSet;
+use std::num::ParseIntError;
+use std::{env, io};
+use std::{path::Path, path::PathBuf, str::FromStr};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct InterpreterFinder {
+    request: InterpreterRequest,
+    sources: SourceSelector,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) enum InterpreterRequest {
+    /// A Python version without an implementation name
+    Version(VersionRequest),
+    /// A path to a directory containing a Python installation
+    Directory(PathBuf),
+    /// A path to a Python executable
+    File(PathBuf),
+    /// The name of a Python executable
+    ExecutableName(String),
```

Comment:
Yes, but I intentionally removed it previously because the request should not need to know _how_ we find the interpreter to fulfill the given executable name.